ca/config.go

```diff
@@ -96,6 +97,22 @@ func (s *SecurityConfig) RootCA() *RootCA {
 	return s.rootCA
 }
 
+// KeyWriter returns the object that can write keys to disk
+func (s *SecurityConfig) KeyWriter() KeyWriter {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	return s.keyReadWriter
```

Comment:
Pointer reads are not guaranteed to be atomic, so if the pointer can change, we should use a lock or `sync/atomic`. However I don't think the pointer can change here.